### PR TITLE
Enable sof-docs deploy in GitHub Actions

### PR DESCRIPTION
**Pull request by xiulipan**
spilt from #321 
depend on #323 as the deploy job needs build job.

### CODE REVIEW

File: .github/actions/create-publish-folder.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# create publish folder for github.io
+mkdir _build/latest
+mv -v _build/html/* _build/latest
+mv _build/latest _build/html/
+touch _build/html/.nojekyll
+mv scripts/publish-README.md _build/html/README.md
+mv scripts/publish-index.html _build/html/index.html
```
Comment (lgirdwood) [inline .github/actions/create-publish-folder.sh:8]: missing an newline here.

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -57,8 +59,41 @@ jobs:
...
+      - name: upload HTML for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: _build/html
+
+  deploy:
+    needs: build-basic
+    runs-on: ubuntu-20.04
```
Comment (lgirdwood) [inline .github/workflows/pull-request.yml:77]: Btw, I'm assuming 20.04 since it's an LTS ?

Comment (marc-hb) [reply .github/workflows/pull-request.yml:77]: I don't think the next one is available on github yet.

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -57,8 +59,41 @@ jobs:
...
+      # https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts
+      - name: upload HTML for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: _build/html
+
+  deploy:
+    needs: build-basic
```
Comment (marc-hb) [inline .github/workflows/pull-request.yml:76]:
```suggestion
    needs: build
```

**Review by marc-hb — APPROVED** [review]
LGTM assuming it works

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -57,8 +59,41 @@ jobs:
...
+      # https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts
+      - name: upload HTML for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: _build/html
+
+  deploy:
+    needs: build
```
Comment (marc-hb) [inline .github/workflows/pull-request.yml:76]:
```suggestion
    needs: supported-reqs
```

**Comment by marc-hb** [comment]
> depend on #323 as the deploy job needs build job.

One-line change, fixed, Github Actions now passing. Please rebase #323 on top of this after it's merged. One-line conflict.

```shell
git fetch origin pull/324/head
git rebase -i HEAD~3 # two fixups
git push -f ... ghdeploy2
```

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (xiulipan) [reply .github/workflows/pull-request.yml:77]: @lgirdwood @marc-hb GitHub Actions support `ubuntu-latest` also, but now the latest is still Ubuntu 20.04 (YES, it is a LTS)
We think stick to a known version will avoid some dependency issue later.

**Comment by xiulipan** [comment]
@lgirdwood Update and fix newline issue.
@marc-hb rebased with the order that we will merge #324 first.
@deb-intel @intelkevinputnam Please help review and merge

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**